### ISSUE TITLE
org.jetbrains.kotlin.js.facade.exceptions.TranslationRuntimeException

### ISSUE THREAD

**Issue by Dominaezzz** [Kotlin/kotlinx.serialization#967]
**Describe the bug**

https://github.com/Dominaezzz/matrix-kt/runs/971422629?check_suite_focus=true#step:8:123

**To Reproduce**

https://github.com/Dominaezzz/matrix-kt/blob/kt-1.4.0/client/src/main/kotlin/io/github/matrixkt/models/events/contents/RoomKeyRequestContent.kt

**Expected behavior**

Successful compilation on JS.

**Environment**
 - Kotlin version: 1.4.0-rc
 - Library version:  1.0.0-1.4.0-rc
 - Kotlin platforms: JS
 - Gradle version: 6.4

**Comment by sandwwraith** [comment]
I think the problematic part is

```kotlin
@Serializer(forClass = RoomKeyRequestContent::class)
    internal object DefaultSerializer
```

The external generation of serializers for sealed classes is not working well

**Comment by TheMrMilchmann** [comment]
I was about to file an issue for this bug right before I found this one. [However, I've already prepared an MCVE, so I might as well link it.](https://github.com/TheMrMilchmann/MCVE/tree/kotlinx-serialization/translation-runtime-exception-custom-polymorphic-serializer)

Additionally, I'd like to point out that the project is building for the JS IR target. This only seems to affect the "legacy" JS target.
